datatrails-common-api/assets/v2/assets/eventresponse.proto

```diff
@@ -173,11 +173,14 @@ message EventResponse {
     // record. We anticipate at least two proof mechs: merkle_log and
     // verkle_log. We use oneof to avoid repeating the scattering of randomly
     // re-purposed fields we currently have for simple hash vs khipu.
-    oneof proof_details {
-        MerkleLogEntry merklelog_entry = 19 [ (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
-            description:
-            "verifiable merkle mmr log entry details"
-            max_length: 1024
-        }];
-    };
+
+
+    // proof details for proof_mechanism MERKLE_LOG
+    MerkleLogEntry merklelog_entry = 19 [ (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
```

Comment:
We have used `oneof` before, in the blockchain API for simplehash details.

Putting as a minor as I don't want to slow down getting this in, but maybe worth taking a look at how we did it there?

Reply:
Ah ok, that is good news. I ran into trouble with mongo. Fixed a number of things and then wasn't keen on another round if I was the first to use oneof. I'll see what I can do over the weekend. Right up until we make this available to customers we can change this. And I like oneof